src/router_core/modules/streaming_link_scrubber/streaming_link_scrubber.c

```diff
@@ -0,0 +1,187 @@
+static int max_free_pool_size = 128;
+
+static void action_handler_CT(qdr_core_t *core, qdr_action_t *action, bool discard);
+
+typedef struct tracker_t tracker_t;
+struct tracker_t {
+    qdr_core_t              *core;
+    qdr_core_timer_t        *timer;
+    qdr_connection_ref_t_sp  next_conn_ref;
+};
+
+
+/* Idle streaming link cleanup
+ *
+ * Check the size of the connections idle link free pool.  If the connection
+ * has accumulated too many unused links start closing them
+ */
+static void idle_link_cleanup(qdr_core_t *core, qdr_connection_t *conn)
+{
+    qdr_link_list_t to_free = DEQ_EMPTY;
+
+    qd_log(core->log, QD_LOG_DEBUG,
+           "[C%"PRIu64"] Streaming link scrubber: scanning connection", conn->identity);
+
+    const size_t pool_size = DEQ_SIZE(conn->streaming_link_pool);
+    if (pool_size > max_free_pool_size) {
+        size_t count = MIN(MAX_FREE_BATCH, pool_size - max_free_pool_size);
+
+        // links are returned to the pool by inserting at the tail.  Thus the
+        // links at head have been on the list the longest and are more likely
+        // be candidates for cleanup (e.g. idle)
+        while (count) {
+            qdr_link_t *link = DEQ_HEAD(conn->streaming_link_pool);
+            if (!qdr_link_is_idle_CT(link))
+                break;
+            DEQ_REMOVE_HEAD_N(STREAMING_POOL, conn->streaming_link_pool);
+            DEQ_INSERT_TAIL_N(STREAMING_POOL, to_free, link);
+            link->in_streaming_pool = false;
+            count -= 1;
+        }
+
+    }
+
+    if (DEQ_HEAD(to_free)) {
+        qd_log(core->log, QD_LOG_DEBUG,
+               "[C%"PRIu64"] Streaming link scrubber: found %d idle links", conn->identity, (int)DEQ_SIZE(to_free));
+
+        while (DEQ_HEAD(to_free)) {
+            qdr_link_t *link = DEQ_HEAD(to_free);
+            DEQ_REMOVE_HEAD_N(STREAMING_POOL, to_free);
+            qd_log(core->log, QD_LOG_DEBUG,
+                   "[C%"PRIu64"][L%"PRIu64"] Streaming link scrubber: closing idle link %s",
+                   link->conn->identity, link->identity, (link->name) ? link->name : "");
+            qdr_link_outbound_detach_CT(core, link, 0, QDR_CONDITION_NONE, true);
+        }
+    }
+}
+
+
+static void timer_handler_CT(qdr_core_t *core, void *context)
+{
+    tracker_t            *tracker    = (tracker_t*) context;
+    qdr_connection_ref_t *first_ref = DEQ_HEAD(core->streaming_connections);
+
+    if (!!first_ref) {
+        qd_log(core->log, QD_LOG_DEBUG, "Starting streaming link scrubber scan");
+        set_safe_ptr_qdr_connection_ref_t(first_ref, &tracker->next_conn_ref);
+        qdr_action_t *action = qdr_action(action_handler_CT, "streaming_link_scrubber");
+        action->args.general.context_1 = tracker;
+        qdr_action_background_enqueue(core, action);
+    } else
+        qdr_core_timer_schedule_CT(core, tracker->timer, timer_interval);
+}
+
+
+static void action_handler_CT(qdr_core_t *core, qdr_action_t *action, bool discard)
```

Comment:
Can we please give this handler a more specific name like qdr_streaming_link_scrubber

Reply:
done!